# Conocicendo los datos

El conjunto de datos incluye la siguiente información:

- Producto y Categoría: Artículos vendidos y sus calificaciones.

- Precio y Envío: Valores de venta y costos asociados.

- Fecha y ubicación de compra: Información temporal y geográfica.

- Evaluación de compra: Comentarios de clientes.

- Tipo de Pago y Cuotas: Métodos utilizados por los clientes.

- Coordenadas Geográficas: Ubicación de las transacciones.

### Líbrerias

In [29]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

### Importación de datos



In [30]:

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda1 = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda1.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

#1. Análisis de facturación



In [39]:
#Calculando el ingreso total de cada tienda
def calcular_ingreso (tienda, num):
  ingreso = tienda.Precio.astype(float).sum()
  print(f'\nEl ingreso total de la tienda {num} es: ${ingreso:.2f}')
  return ingreso

ingreso1 = calcular_ingreso(tienda1, 1)
ingreso2 = calcular_ingreso(tienda2, 2)
ingreso3 = calcular_ingreso(tienda3, 3)
ingreso4 = calcular_ingreso(tienda4, 4)



El ingreso total de la tienda 1 es: $1150880400.00

El ingreso total de la tienda 2 es: $1116343500.00

El ingreso total de la tienda 3 es: $1098019600.00

El ingreso total de la tienda 4 es: $1038375700.00


In [46]:
#Se crea DataFrame Ingresos para poder graficar
ingresos = {
    'Tienda': ['Tienda 1', 'Tienda 2', 'Tienda 3', 'Tienda 4'],
    'Ingreso_Total': [ingreso1, ingreso2, ingreso3, ingreso4]
}

df_ingresos = pd.DataFrame(ingresos)

In [61]:
#Gráfico de barras
fig = px.bar(df_ingresos,
             x='Tienda',
             y='Ingreso_Total',
             title='Ingreso total por tienda',
             labels={'Ingreso_Total': 'Ingreso ($)', 'Tienda': 'Tienda'},
             text='Ingreso_Total')
fig.update_traces(texttemplate='$%{text:.2f}', textposition='outside')
fig.show()

# 2. Ventas por categoría

In [52]:
tiendas = [tienda1, tienda2, tienda3, tienda4]

for i, tienda in enumerate(tiendas, 1):
    ventas = tienda.groupby('Categoría del Producto')['Precio'].sum().round(2).sort_values(ascending=False).reset_index()

    # Imprimir resultados
    print(f"\nVentas por categoría en Tienda {i}:\n")
    print(ventas)

    # Graficar
    fig = px.bar(ventas,
                 x='Categoría del Producto',
                 y='Precio',
                 title=f'Ventas por categoría - Tienda {i}',
                 labels={'Precio': 'Ventas ($)', 'Categoría del Producto': 'Categoría'},
                 text='Precio')
    fig.update_traces(texttemplate='$%{text:.2f}', textposition='outside')
    fig.update_layout(xaxis_tickangle=-45)
    fig.show()


Ventas por categoría en Tienda 1:

    Categoría del Producto       Precio
0             Electrónicos  429493500.0
1        Electrodomésticos  363685200.0
2                  Muebles  187633700.0
3   Instrumentos musicales   91299000.0
4     Deportes y diversión   39290000.0
5                 Juguetes   17995700.0
6  Artículos para el hogar   12698400.0
7                   Libros    8784900.0



Ventas por categoría en Tienda 2:

    Categoría del Producto       Precio
0             Electrónicos  410831100.0
1        Electrodomésticos  348567800.0
2                  Muebles  176426300.0
3   Instrumentos musicales  104990300.0
4     Deportes y diversión   34744500.0
5                 Juguetes   15945400.0
6  Artículos para el hogar   14746900.0
7                   Libros   10091200.0



Ventas por categoría en Tienda 3:

    Categoría del Producto       Precio
0             Electrónicos  410775800.0
1        Electrodomésticos  329237900.0
2                  Muebles  201072100.0
3   Instrumentos musicales   77380900.0
4     Deportes y diversión   35593100.0
5                 Juguetes   19401100.0
6  Artículos para el hogar   15060000.0
7                   Libros    9498700.0



Ventas por categoría en Tienda 4:

    Categoría del Producto       Precio
0             Electrónicos  409476100.0
1        Electrodomésticos  283260200.0
2                  Muebles  192528900.0
3   Instrumentos musicales   75102400.0
4     Deportes y diversión   33350100.0
5                 Juguetes   20262200.0
6  Artículos para el hogar   15074500.0
7                   Libros    9321300.0


# 3. Calificación promedio de la tienda


In [54]:
#Calcular las calificaciones promedio de los clientes para cada tienda
def calificacion_promedio(tienda, num):
  calificacion = tienda['Calificación'].mean().round(2)
  print(f'\nLa calificación promedio de la tienda {num} es: {calificacion}')
  return calificacion

calificacion1 = calificacion_promedio(tienda1, 1)
calificacion2 = calificacion_promedio(tienda2, 2)
calificacion3 = calificacion_promedio(tienda3, 3)
calificacion4 = calificacion_promedio(tienda4, 4)


La calificación promedio de la tienda 1 es: 3.98

La calificación promedio de la tienda 2 es: 4.04

La calificación promedio de la tienda 3 es: 4.05

La calificación promedio de la tienda 4 es: 4.0


In [57]:
#Se crea DataFrame calificacion para poder graficar
calificacion = {
    'Tienda': ['Tienda 1', 'Tienda 2', 'Tienda 3', 'Tienda 4'],
    'Calificacion Promedio': [calificacion1 , calificacion2 , calificacion3 , calificacion4 ]
}

df_calificacion  = pd.DataFrame(calificacion)

In [60]:
#Gráfico de barras
fig = px.bar(df_calificacion,
             x='Tienda',
             y='Calificacion Promedio',
             title='Calificacion Promedio por tienda',
             labels={'Calificacion Promedio': 'Calificacion', 'Tienda': 'Tienda'},
             text='Calificacion Promedio')
fig.update_traces(texttemplate='%{text:.2f}', textposition='outside')
fig.show()

# 4. Productos más y menos vendidos

In [37]:
def productos_popular(tienda, num):
  cantidad = tienda.Producto.value_counts()
  precios = tienda.groupby('Producto')['Precio'].first()
  df = pd.DataFrame({'Cantidad': cantidad, 'Precio': precios})
  df['Ingreso'] = df['Cantidad'] * df['Precio']
  df_ordenado = df.sort_values(by='Ingreso', ascending=False)
  top =  df_ordenado.head(3)
  bottom =  df_ordenado.tail(3)

  print(f"Tienda {num} - TOP 3 productos por ingreso:\n{top}\n")
  print(f"Tienda {num} - BOTTOM 3 productos por ingreso:\n{bottom}\n")
  print('---------------------------------------------------')

productos_popular(tienda1, 1)
productos_popular(tienda2, 2)
productos_popular(tienda3, 3)
productos_popular(tienda4, 4)


Tienda 1 - TOP 3 productos por ingreso:
               Cantidad     Precio      Ingreso
Producto                                       
Refrigerador         54  2431300.0  131290200.0
Iphone 15            49  2525400.0  123744600.0
TV LED UHD 4K        60  1891700.0  113502000.0

Tienda 1 - BOTTOM 3 productos por ingreso:
                    Cantidad   Precio   Ingreso
Producto                                       
Cubo mágico 8x8           44  17100.0  752400.0
Cuerda para saltar        40  18400.0  736000.0
Dinosaurio Rex            40  13900.0  556000.0

---------------------------------------------------
Tienda 2 - TOP 3 productos por ingreso:
               Cantidad     Precio      Ingreso
Producto                                       
Iphone 15            55  2580500.0  141927500.0
TV LED UHD 4K        45  2659900.0  119695500.0
Refrigerador         49  1817600.0   89062400.0

Tienda 2 - BOTTOM 3 productos por ingreso:
                    Cantidad   Precio   Ingreso
Producto   

# 5. Envío promedio por tienda

In [62]:
#Calcular el costo de envío promedio para cada tienda
def envio_promedio(tienda, num):
  envio = tienda['Costo de envío'].mean().round(2)
  print(f'\nEl costo de envío promedio de la tienda {num} es: ${envio}')
  return envio

envio1 = envio_promedio(tienda1, 1)
envio2 = envio_promedio(tienda2, 2)
envio3 = envio_promedio(tienda3, 3)
envio4 = envio_promedio(tienda4, 4)


El costo de envío promedio de la tienda 1 es: $26018.61

El costo de envío promedio de la tienda 2 es: $25216.24

El costo de envío promedio de la tienda 3 es: $24805.68

El costo de envío promedio de la tienda 4 es: $23459.46


In [64]:
#Se crea DataFrame costo de envio para poder graficar
envios = {
    'Tienda': ['Tienda 1', 'Tienda 2', 'Tienda 3', 'Tienda 4'],
    'Costo del envio Promedio': [envio1 , envio2 , envio3 , envio4 ]
}

df_envios  = pd.DataFrame(envios)

In [66]:
#Gráfico de barras
fig = px.bar(df_envios,
             x='Tienda',
             y='Costo del envio Promedio',
             title='Costo del envio Promedio por tienda',
             labels={'Costo del envio Promedio': 'Costo del envio Promedio', 'Tienda': 'Tienda'},
             text='Costo del envio Promedio')
fig.update_traces(texttemplate='$%{text:.2f}', textposition='outside')
fig.show()

### Informe final

#### Introducción

El presente informe tiene como objetivo recomendar la mejor tienda para que el Sr. Juan venda sus productos, basándose en los datos actualizados de ingresos totales, calificaciones promedio de los clientes y costos de envío. A diferencia del análisis inicial, los nuevos datos revelan un panorama de rendimiento muy diferente para cada una de las tiendas.

#### Análisis de Datos Actualizados
1. Ingresos Totales por Tienda
Los nuevos datos muestran un claro líder en términos de volumen de ventas:

- Tienda 1: Es la que genera los mayores ingresos, con $1,150,880,400.00.

- Tienda 2: Le sigue con $1,116,343,500.00.

- Tienda 3: Se encuentra en un tercer lugar cercano, con $1,098,019,600.00.

-  Tienda 4: Es la que tiene los ingresos más bajos, con $1,038,375,700.00.

En términos de potencial de ganancias, la Tienda 1 es la opción más fuerte, con una ventaja significativa sobre las demás.

2. Calificaciones Promedio de los Clientes
La satisfacción del cliente presenta un orden inverso al de los ingresos:

- Tienda 3: Tiene la calificación más alta, con 4.05.

- Tienda 2: Le sigue de cerca con 4.04.

- Tienda 4: Se ubica en tercer lugar con 4.00.

- Tienda 1: Es la tienda con la calificación promedio más baja, con 3.98.

Este dato es crucial, ya que sugiere que, a pesar de sus altos ingresos, la Tienda 1 tiene problemas de satisfacción del cliente, lo que podría afectar la retención a largo plazo. Por el contrario, la Tienda 3, aunque con menores ingresos, destaca en la calidad de su servicio o productos.

3. Costo de Envío Promedio
El costo de envío es un factor clave para la rentabilidad. Los datos actualizados muestran que la Tienda 4 es la más eficiente en este aspecto:

- Tienda 4: El costo de envío promedio más bajo, con $23,459.46.

- Tienda 3: Le sigue con $24,805.68.

- Tienda 2: Presenta un costo de envío promedio de $25,216.24.

- Tienda 1: Es la tienda con el costo de envío más alto, con $26,018.61.

La Tienda 1, a pesar de su alto volumen de ventas, incurre en los mayores costos de envío, lo que podría impactar su margen de beneficio.

#### Conclusión y Recomendación Final
La decisión ya no es tan simple, ya que ninguna tienda sobresale en todos los aspectos. Se deben considerar las fortalezas y debilidades de cada una:

- Tienda 1: Es la opción para maximizar el volumen de ventas y los ingresos totales, pero a costa de la satisfacción del cliente y con los costos de envío más altos.

- Tienda 3: Es la mejor opción si la prioridad es la reputación y la satisfacción del cliente, ya que tiene la calificación más alta y costos de envío bajos.

- Tienda 4: Es la más eficiente en términos de costos de envío, pero su menor volumen de ingresos la hace menos atractiva en términos de ganancias totales.

#### Recomendación:

A pesar de las desventajas en satisfacción y costos, se recomienda al Sr. Juan que elija la Tienda 1 como su principal socio.

#### Justificación:

La ventaja de más de $30 millones en ingresos totales sobre su competidor más cercano (la Tienda 2) es un factor determinante para el crecimiento y la rentabilidad. La Tienda 1 es, sin lugar a dudas, el mercado con mayor demanda. Aunque las calificaciones bajas y los costos de envío altos son un riesgo, pueden ser mitigados. El Sr. Juan puede aprovechar el alto volumen de ventas de la Tienda 1, y al mismo tiempo, trabajar en mejorar la calidad de sus productos y la logística para aumentar las calificaciones y negociar mejores términos de envío.

En resumen, la Tienda 1 ofrece el mayor potencial de ganancias, y los desafíos asociados pueden ser gestionados estratégicamente.